sdk/identity/Azure.Identity/src/Credentials/WorkloadIdentityCredential.cs

```diff
@@ -47,6 +47,18 @@ public WorkloadIdentityCredential(WorkloadIdentityCredentialOptions options)
                 clientAssertionCredentialOptions.Pipeline = options.Pipeline;
                 clientAssertionCredentialOptions.MsalClient = options.MsalClient;
 
+                // Configure Kubernetes token proxy if user opted in
+                if (options.AzureKubernetesTokenProxy)
+                {
+                    var proxyConfig = KubernetesProxyConfig.TryCreate();
+                    if (proxyConfig != null)
```

Comment:
If it is null, should we throw?

Reply:
Discussed offline - the proxy isn't required for authentication, so we can't assume the credential won't be able to authenticate. A developer may deploy the same code with and without an identity binding